File: 1/rockets.py
def fuel_mass(mass):
    return mass//3 - 2

def masses_iterator(masses):
    for m in masses:
        fuel = fuel_mass(m)
        while fuel > 0:
            yield fuel
            fuel = fuel_mass(fuel)

masses = [int(l.rstrip("\n")) for l in open("input.txt").readlines()]

print("Part 1:", sum([fuel_mass(m) for m in masses]))
print("Part 2:", sum(masses_iterator(masses)))

File: 01/rockets.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <cmath>

int calculate_fuel(int mass) {
	return std::floor((float)mass/3-2);
}

int main() {
	std::ifstream f("input.txt");
	std::vector<int> masses;
	std::string line;
	while(std::getline(f, line)) {
		if (line.size() > 0) {
			masses.push_back(std::stoi(line));
		}
	}

	int part_one{}, part_two{}, fuel{};
	for (auto const& mass: masses) {
		fuel = calculate_fuel(mass);
		part_one += fuel;
		while (fuel > 0) {
			part_two += fuel;
			fuel = calculate_fuel(fuel);
		}
	}
	std::cout << "Part 1: " << part_one << '\n';
	std::cout << "Part 2: " << part_two << std::endl;
}
